spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,233 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.spark.connector;
+
+import com.datastrato.gravitino.dto.rel.partitioning.Partitioning.SingleFieldPartitioning;
+import com.datastrato.gravitino.rel.expressions.Expression;
+import com.datastrato.gravitino.rel.expressions.NamedReference;
+import com.datastrato.gravitino.rel.expressions.distributions.Distribution;
+import com.datastrato.gravitino.rel.expressions.distributions.Distributions;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrder;
+import com.datastrato.gravitino.rel.expressions.sorts.SortOrders;
+import com.datastrato.gravitino.rel.expressions.transforms.Transform;
+import com.datastrato.gravitino.rel.expressions.transforms.Transforms;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.ws.rs.NotSupportedException;
+import lombok.Getter;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.spark.sql.connector.expressions.BucketTransform;
+import org.apache.spark.sql.connector.expressions.Expressions;
+import org.apache.spark.sql.connector.expressions.IdentityTransform;
+import org.apache.spark.sql.connector.expressions.LogicalExpressions;
+import org.apache.spark.sql.connector.expressions.SortedBucketTransform;
+import scala.collection.JavaConverters;
+
+/**
+ * SparkTransformConverter translate between Spark transform and Gravitino partition, distribution,
+ * sort orders. There may be multi partition transforms, but should be only one bucket transform.
+ * Spark bucket transform is corresponding to Gravitino Hash distribution without sort orders. Spark
+ * sorted bucket transform is corresponding to Gravitino Hash distribution with sort orders.
+ */
+public class SparkTransformConverter {
```

Comment:
Why do we split this into three converters: `partitioningConverter`, `distributionConverter` and `sortOrderConverter`.  This corresponds to three different scenarios and will make the code much clearer